twine/auth.py

```diff
@@ -0,0 +1,80 @@
+import warnings
+import getpass
+import functools
+
+import keyring
+import munch
+
+from . import utils
+from . import exceptions
+
+
+class Resolver:
+    def __init__(self, settings, username=None, password=None):
+        self.settings = settings
+        self.input = munch.Munch(username=username, password=password)
+
+    @property
```

Comment:
This seems to be the recommended workaround for this [open mypy issue](https://github.com/python/mypy/issues/1362):

```suggestion
    @property  # type: ignore  # https://github.com/python/mypy/issues/1362
```

Athough we might also want to be aware of [lru_cache annotation doesn't work as a property · Issue #5858](https://github.com/python/mypy/issues/5858).

That said, this feels like an implementation of `cached_property` from [Django](https://docs.djangoproject.com/en/2.2/ref/utils/#django.utils.functional.cached_property), [Python 3.8](https://docs.python.org/3.8/library/functools.html#functools.cached_property), and at least one [standalone library](https://pypi.org/project/cached-property/). Is it worth implementing something like that here?